### DISCUSSION TITLE
How to display images from Firestore on FlutterFlow using Rowy?

### DISCUSSION THREAD

**Discussion by BeeBombshell** [AntlerVC/firetable#1085] [Support / Q&A] [answered]
I am using Rowy as a CMS to manage all the data to be displayed on my FlutterFlow project. But I'm not sure how to make an image appear in FlutterFlow's Photo Path data column.

**Comment by BeeBombshell** [comment, marked as answer]
We can use the Derivative Field to extract the `downloadURL` derived from an image uploaded in another column. Refer to the following video to see how to get the download URL.

https://user-images.githubusercontent.com/53828745/213441093-dd4a0695-e960-407f-8d88-121c1472c6dc.mp4

### STEPS:
1. Create a Derivative field and set the **Listener fields** to be the **Image** column.
2. Change the **Output field** type to **URL**.
3. Set the Derivative script to this:
```js
const derivative:Derivative = async ({row,ref,db,storage,auth,logging})=>{
    let url = row.image[0].downloadURL;
    return url;
  }
```
4. Continue to add images and retrieve the URL in the derivative column.

Now, you can continue to use the derivative column as an Image Path in FlutterFlow.

**Reply by DayneLinx** [reply]
This looks promising, thank you. I will try this once I figure out why I can no longer display data in a listview.

**Reply by DayneLinx** [reply]
Thanks @BeeBombshell! That worked. How do I collect multiple images as this works for only one? When I upload multiple images it only references the first in the index [0]. How do I reference multiple images as I am creating a gallery.

**Reply by BeeBombshell** [reply]
Hey @DayneLinx. To access multiple image URLs, you could create another **derivative** column. Set it's **Listener Fields** to the Image field and the **Output Field Type** to **JSON**. Now can paste the following to the derivative script.

```js
const derivative:Derivative = async ({row,ref,db,storage,auth,logging})=>{
  // WRITE YOUR CODE ONLY BELOW THIS LINE. DO NOT WRITE CODE/COMMENTS OUTSIDE THE FUNCTION BODY
  const obj = {
    data: []
  }
  row.image.map((img) => {
    obj.data.push(img.downloadURL)
  })
  return obj;
}
```

### Output

<img width="1512" alt="image" src="https://user-images.githubusercontent.com/53828745/216257124-dc1d90e8-a80a-45c5-ad87-31851f93b19c.png">

Hope this helps! :)
